internal/compiler/langtype.rs

```diff
@@ -481,6 +481,27 @@ impl ElementType {
         }
     }
 
+    pub fn can_have_child(&self, name: &str, tr: &TypeRegister) -> bool {
```

Comment:
What is the difference with lookup_type_for_child_element?
This looks like you could just call `lookup_type_for_child_element(name, tr).is_ok()`. What am i missing?
Is it for performance reasons?
If possible, would be nice to not duplicate the logic.

Reply:
Ah yes indeed.

I can think of the following option.
 1. keep it as is
 2. split lookup_type_for_child_element in two and have one part call the other
 3. make lookup_type_for_child_element return an LookupChildElementError enum instead of a string and in the LSP ignore the LookupChildElementError::UnkownElement one.

No strong opinion on this. I'd have a slight preference for 2. if i was doing it myself, But i'd be ok with 1.